ivy/utils/backend/sub_backend_handler.py

```diff
@@ -0,0 +1,204 @@
+import os
+from types import ModuleType, FunctionType
+import logging
+import importlib
+
+import ivy
+from ivy.func_wrapper import _wrap_function
+from ivy.utils.exceptions import IvyException
+
+
+_backends_subpackage_path = "ivy.functional.backends"
+_sub_backend_dict = dict()
+_backend_to_sub_backends_dict = dict()
+
+
+# dynamic sub_backend detection
+for backend in os.listdir(
+    os.path.join(
+        ivy.__path__[0].rpartition(os.path.sep)[0],  # type: ignore
+        _backends_subpackage_path.replace(".", os.path.sep),
+    )
+):
+    if backend.startswith("__"):
+        continue
+
+    sub_backends_dir = os.path.join(
+        ivy.__path__[0].rpartition(os.path.sep)[0],
+        _backends_subpackage_path.replace(".", os.path.sep),
+        backend,
+        "sub_backends",
+    )
+    if os.path.isdir(sub_backends_dir):
```

Comment:
Is this check here necessary?

Reply:
oh this was there before we decided to remove the `try and except` imports block, now it's useless, good catch! will get it removed:)